renderite/serialization.zig

```diff
@@ -82,10 +131,32 @@ pub const IpcSerializer = struct {
         try self.writer.writeInt(T, value, endian);
     }
 
+    pub fn writeFloat(self: IpcSerializer, comptime T: type, value: T) !void {
+        try self.writer.writeAll(std.mem.asBytes(&value));
+    }
+
+    pub fn writeEnum(self: IpcSerializer, comptime T: type, value: T) !void {
+        try self.writeInt(@typeInfo(T).@"enum".tag_type, @intFromEnum(value));
+    }
+
+    pub fn writeBool(self: IpcSerializer, value: bool) !void {
+        try self.writeInt(u8, if (value) 1 else 0);
+    }
```

Comment:
`u1` coerces to `u8`

```suggestion
    pub fn writeBool(self: IpcSerializer, value: bool) !void {
        try self.writeInt(u8, @intFromBool(value));
    }
```